### PR TITLE
added wrapper

### PR DESCRIPTION
**Pull request by ajstanley**

### CODE REVIEW

File: includes/emicdora.inc [hunk 1]
```diff
@@ -26,10 +27,31 @@ function collationtools_get_edited_collation($document_id) {
...
-
+  $raw_contents = file_get_contents($url);
+  $contents = str_replace($br_token, '<br />', $raw_contents);
+  $dom = new DOMDocument();
+  $dom->loadHTML($contents);
+  $bodies = $dom->getElementsByTagName('body');
+  $body = $bodies->item(0);
+  // Get rid of comment nodes.
+  foreach ($dom->childNodes as $node) {
+    if ($node->nodeName == '#comment') {
```
Comment (adam-vessey) [inline includes/emicdora.inc]: Probably safer to compare the type of the node... As in, something like:
```php
if ($node->nodeType === XML_COMMENT_NODE) {
```

File: includes/emicdora.inc [hunk 2]
```diff
@@ -26,10 +27,31 @@ function collationtools_get_edited_collation($document_id) {
...
+      $node->parentNode->removeChild($node);
+    }
+  }
+  $all_nodes = $body->childNodes;
+  foreach ($all_nodes as $node) {
+    // Convert text nodes to spans.
+    if ($node->nodeName == '#text') {
```
Comment (adam-vessey) [inline includes/emicdora.inc]: Same as https://github.com/discoverygarden/austese_collation/pull/25/files#r22484543, WRT type checking.

File: includes/emicdora.inc [hunk 3]
```diff
@@ -26,10 +27,31 @@ function collationtools_get_edited_collation($document_id) {
       'version2' => $_GET['version2'],
       'diff_kind' => $_GET['diff_kind'],
   )));
-  $contents = file_get_contents($url);
-  $spans = preg_split( '#</span>#', $contents, -1, PREG_SPLIT_DELIM_CAPTURE | PREG_SPLIT_NO_EMPTY);
-  $fixed_content = collationtools_fix_calliope($spans, $_GET['diff_kind']);
-
+  $raw_contents = file_get_contents($url);
+  $contents = str_replace($br_token, '<br />', $raw_contents);
```
Comment (ajstanley) [inline includes/emicdora.inc]: The austese engine provides us with broken xml - it splits up the br tags.  This was the workaround to make it all work for us.

File: includes/emicdora.inc [hunk 4]
```diff
@@ -26,10 +27,31 @@ function collationtools_get_edited_collation($document_id) {
...
+  $dom = new DOMDocument();
+  $dom->loadHTML($contents);
+  $bodies = $dom->getElementsByTagName('body');
+  $body = $bodies->item(0);
+  // Get rid of comment nodes.
+  foreach ($dom->childNodes as $node) {
```
Comment (ajstanley) [inline includes/emicdora.inc]: Not necessary - there's only ever one comment per MVD - this is just to clear the unused css out of the input.

File: includes/emicdora.inc [hunk 5]
```diff
@@ -26,10 +27,31 @@ function collationtools_get_edited_collation($document_id) {
...
+    if ($node->nodeName == '#comment') {
+      $node->parentNode->removeChild($node);
+    }
+  }
+  $all_nodes = $body->childNodes;
+  foreach ($all_nodes as $node) {
```
Comment (ajstanley) [inline includes/emicdora.inc]: Not needed,  The text nodes shouldn't be there at all.  This is just there to get orphaned text nodes and wrap them in spans are they were meant to be.

File: includes/emicdora.inc (hunk 4 above)
Comment (adam-vessey) [inline includes/emicdora.inc]: Could use a comment to this effect inline...

File: includes/emicdora.inc (hunk 5 above)
Comment (adam-vessey) [inline includes/emicdora.inc]: Could use a comment to this effect inline.

File: includes/emicdora.inc [hunk 6]
```diff
@@ -26,10 +27,33 @@ function collationtools_get_edited_collation($document_id) {
...
+  // Get rid of comment nodes.
+  foreach ($dom->childNodes as $node) {
+    if ($node->nodeType === XML_COMMENT_NODE) {
+      $node->parentNode->removeChild($node);
+    }
+  }
+  $all_nodes = $body->childNodes;
+  foreach ($all_nodes as $node) {
+    // Convert text nodes to spans.
+    if ($node->nodeType == XML_TEXT_NODE) {
```
Comment (adam-vessey) [inline includes/emicdora.inc]: Not a huge deal: Should probably be consistent with the operator... `==` vs `===`. (`===` is used for comments above)

**Pull request merged**